docs/design/deployment_design.md

```diff
@@ -0,0 +1,218 @@
+
+  Arktos running in AWS. It has one master node and multiple worker nodes.
+
+- Arktos GCP environment - multi master
+
+  Arktos running in GCP. It has multiple API servers and multiple ETCD. Consider it has multiple master nodes.
+
+- Arktos GCP environment with Kubemark
+
+  Arktos running in GCP. It uses Kubemark to simulate large clusters with hundreds of nodes. It's a good environment to demonstrate Mizar performance in large clusters.
+
+### Dev vs Prod of Mizar
+
+All the time these should be two version of Mizar to be deployed. One is dev version and one is prod version.
+
+- **Prod Version**, it's for end users to use Mizar in their test or prod environment. Its purpose is not to test Mizar. Instead it's to test their environment with Mizar, or check how Mizar fit their requirements, or using Mizar in their production environment. It's to deploy Mizar stable versions. We shall provide public docker images for prod version deployment. 
+
+- **Dev Version**, it always reflects latest code of Mizar. The code can be from any branch such as master, dev-next or any private branch. In this way, while developing Mizar, developers can easily test Mizar by latest code. It's for Mizar developers with dev or test purpose. 
+
+We plan to add continuous deployment into a test cluster. We will use Jenkins for Mizar CI/CD scenario which will make Mizar more robust.
+
+### New Install vs Update
+
+The deploy script should support both new install and update of Mizar. In prod environment, when there is Mizar installed, user may want to update Mizar to new version while cluster networking won't break. We need to support "Mizar update" scenario. While Mizar updating, there should be no impact or minimal/acceptable impact to the cluster.
+
+### CI/CD
+
+We planned Mizar CI/CD pipeline, and it will be for both prod and dev versions of Mizar. The CI/CD pipeline will test Mizar update scenario continuously.
+
+Jenkins is a good candidate tool for CI/CD. What we planned are:
+
+- After send pull request, unit/integration/e2e test will be launched to verify there is no regression for new code. After verification succeed, code could be merged. (CI)
+
+- We will implement perf test and add it to CI. This is long term target which won't be in iteration 1. Depends on perf test cost, later we may decide it's not in CI which will be triggered in each pull request. Instead we may put it triggered by merge, which will reduce frequence of occur times. (CI)
+
+- After code merged to dev-next branch, a job will be triggered to build binaries, docker images. A test cluster should be deployed and e2e test will be executed. (CD)
+
+- After code merged to master branch, a job will be triggered to build prod binaries, prod docker images. Then end user can use the latest prod build to deploy. (CD)
+
+### Deploy Steps
+
+1. Binary Ready
+
+	In the first step we need to make sure all Mizar components are in a binary ready state for deployment. 
```

Comment:
We currently do this manually. (Manually compile Mizar, build docker image, and push to the official Futurewei dockerhub account fwnetworking (requires login)).
We will need to keep these images updated for users to be able to easily setup mizar in their own  k8s environments.
How shall we automate this?

Reply:
CI/CD part has a description that build production when code merged to master.